File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    map<string, int> index;
    for (int i = 0; i < want.size(); i++)
        index[want[i]] = i;

    for (int i = 0; i <= discount.size()-10; i++) {
        vector<int> nums = vector<int>(number.size());
        for (int j = i; j < i + 10 && j < discount.size(); j++) {
            if (index.find(discount[j]) == index.end()) break;
            nums[index[discount[j]]]++;
        }

        bool flag = true;
        for (int j = 0; j < number.size(); j++) {
            if (number[j] != nums[j]) {
                flag = false;
                break;
            }
        }
        if (flag) answer++;
    }
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.swift
import Foundation

func solution(_ want:[String], _ number:[Int], _ discount:[String]) -> Int {
    var answer: Int = 0
    let number = Dictionary<String, Int>(uniqueKeysWithValues: zip(want, number))
    
    for i in 0...(discount.count-10) {
        var count = number
        
        for j in 0..<10 {
            if want.contains(discount[i+j]) {
                count[discount[i+j]]! -= 1
            } else {
                break 
            }
        }
        
        if count.filter { $0.value == 0 }.count == count.count {
            answer += 1
        }
    }
    
    return answer
}